main.go

```diff
@@ -35,13 +42,264 @@ func main() {
+	expectedResults := 0
+	for _, file := range diffFiles {
+		go parseFileDiff(rootFolder+file.Name(), resChannel)
+		expectedResults++
+	}
+
+	diffResult := diffResult{
+		functionCalls: make(map[string]int),
+	}
+
+	// Get results from the communication channel
+	for i := 0; i < expectedResults; i++ {
+		routineRes := <-resChannel
+
+		diffResult.regions += routineRes.regions
+		diffResult.lineAdded += routineRes.lineAdded
+		diffResult.lineDeleted += routineRes.lineDeleted
+		diffResult.files = append(diffResult.files, routineRes.files...)
+
+		for key, value := range routineRes.functionCalls {
+			diffResult.functionCalls[key] += value
+		}
+	}
+
+	return &diffResult
+}
+
+// Parse the content of a diff file and send the result through the channel
+func parseFileDiff(filename string, resChannel chan diffResult) {
+	// Define all the regex that we need to parse the Diffs
+	var regionPrefix = "^@@"
+	var diffCallPrefix = "diff --git "
+	var diffCallRegex = regexp.MustCompile(regexp.QuoteMeta(diffCallPrefix))
+	var deletedPrefix = "^" + regexp.QuoteMeta("- ")
+	var addedPrefix = "^" + regexp.QuoteMeta("+ ")
+
+	// Parsing result and parsing variables
+	res := diffResult{
+		functionCalls: make(map[string]int),
+	}
+
+	var matched bool
+	var regexError error
+
+	// Open the file the read line by line
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if err := scanner.Err(); err != nil {
+			log.Fatal(err)
+		}
+
+		// Find regions blocks
+		matched, regexError = regexp.MatchString(regionPrefix, line)
+		if matched && regexError == nil {
+			res.regions++
+			continue
+		}
+
+		// Find file names in diff call
+		matched, regexError = regexp.MatchString(diffCallPrefix, line)
+		if matched && regexError == nil {
+			var files = diffCallRegex.ReplaceAllString(line, "")
+			res.files = append(res.files, strings.Split(files, " ")...)
+			continue
+		}
+
+		// Find deleted lines
+		matched, regexError = regexp.MatchString(deletedPrefix, line)
+		if matched && regexError == nil {
+			res.lineDeleted++
+			continue
+		}
+
+		// Find added lines
+		matched, regexError = regexp.MatchString(addedPrefix, line)
+		if matched && regexError == nil {
+			res.lineAdded++
+
+			// checks if the added line contains a method call
+			for _, call := range *extractMethodCalls(line) {
+				if strings.Trim(call, " ") != "" {
+					res.functionCalls[call]++
+				}
+			}
+
+			continue
+		}
+	}
+
+	// Send result back to the main routine and close the read file
+	resChannel <- res
+
+	err = file.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Extract method calls from a given diff line
+func extractMethodCalls(line string) *[]string {
+	// RegEx to find function and method calls
+	functionRegex := regexp.MustCompile("[a-zA-Z_]+" + regexp.QuoteMeta(".") + "?" + "[a-zA-Z]+" + regexp.QuoteMeta("(") + ".*" + regexp.QuoteMeta(")"))
```

Comment:
This will work on `x.y(asd)` but not on `x.y.z(asd)`, and I stand my point, the `QuoteMeta` hinder readability for your regexps